src/main/java/seedu/address/model/job/Job.java

```diff
@@ -0,0 +1,116 @@
+package seedu.address.model.job;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.person.Person;
+
+/**
+ * Represents a job. Immutable.
+ */
+public class Job {
+
+    private final String name;
+    private final Money rate;
+    private final float duration;
+
+    private final boolean isPaid;
+
+    private final Set<Person> persons = new HashSet<>();
+
+    /**
+     * Constructor for an immutable job.
+     * All fields must not be null.
+     */
+    public Job(String name, Money rate, float duration, boolean isPaid, Set<Person> persons) {
```

Comment:
Perhaps can use `hasBeenPaid` instead of `isPaid` for better understandability. Both cases adhere to coding standard tho

Reply:
Noted, I think I will use `hasPaid`.